server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -98,6 +98,20 @@ package object server {
       }
     }
 
+    def noSpiderWithEither[F[_]: Monad, E, T](
+        authUserWithError: Kleisli[EitherT[F, E, ?], Request[F], T],
```

Comment:
This is tantalizingly close to something we can express in terms of `MonadError`, but I think we'd need a `G[_]` parameter and a natural transformation `G ~> F` and maybe even back `F ~> G`.  I can't quite make the types click in my head.

Reply:
Thanks for reviewing, Ross. I'm having trouble implementing this suggestion. So I know what the `F` parameter represents, but please let me know the need for the `G`.

Reply:
Well, looking at https://github.com/http4s/http4s/blob/master/server/src/main/scala/org/http4s/server/package.scala#L104-L105, should I update my PR be consistent, namely changing:

>         authUserWithError: Kleisli[EitherT[F, E, ?], Request[F], T],

to

>        authUserWithError: Kleisli[F, Request[F], Either[E, T],

?

Reply:
I was thinking `Kleisli[G, Request[F], T]`.  I think it would require a `G ~> F` because we don't have a monad transformer typeclass to relate the `G` and `F`.  It might not be worth it.

Reply:
ACK, @rossabaker. Do you think that this change is useful?

I'm asking because it still seems to be a hassle to specify the path in the `val authUserWithError: Kleisli[EitherT[IO, Error, ?], Request[IO], User]`, per my spec, https://github.com/http4s/http4s/pull/2585/commits/a60b2496c7dd8e1ffec92dfc377c24a0ff9ae175/.

It seems simplest to me to auth all requests, regardless of the path, and then return an HTTP-404/401 if that check fails.

As an aside, I'd be interested in the implementation of https://github.com/http4s/http4s/pull/2585#discussion_r287572967!